test/k8sT/assertionHelpers.go

```diff
@@ -38,18 +38,15 @@ func ExpectKubeDNSReady(vm *helpers.Kubectl) {
 // ExpectCiliumReady is a wrapper around helpers/WaitForPods. It asserts that
 // the error returned by that function is nil.
 func ExpectCiliumReady(vm *helpers.Kubectl) {
-	err := vm.WaitforPods(helpers.KubeSystemNamespace, "-l k8s-app=cilium", longTimeout)
-	ExpectWithOffset(1, err).Should(BeNil(), "cilium was not able to get into ready state")
-
-	err = vm.CiliumPreFlightCheck()
+	ExpectCiliumRunning(vm)
+	err := vm.CiliumPreFlightCheck()
 	ExpectWithOffset(1, err).Should(BeNil(), "cilium pre-flight checks failed")
 }
 
 // ExpectCiliumOperatorReady is a wrapper around helpers/WaitForPods. It asserts that
 // the error returned by that function is nil.
 func ExpectCiliumOperatorReady(vm *helpers.Kubectl) {
-	err := vm.WaitforPods(helpers.KubeSystemNamespace, "-l name=cilium-operator", longTimeout)
-	ExpectWithOffset(1, err).Should(BeNil(), "Cilium operator was not able to get into ready state")
+	ExpectDeployReady(vm, "kube-system", "cilium-operator", longTimeout)
```

Comment:
This PR has a lot of changes - which may make it hard to debug if it fails - maybe tease out stuff into separate commits and add the base infrastructure along with adding something like `ExpectCIliumOperatorReady` as the initial consumer of `ExpectDeployReady` since it's used in all K8s tests? This isn't strictly necessary, but may make your life easier for getting this merged since this commit contains a lot changes.